tests/tests_app/utilities/test_network.py

```
@@ -1,9 +1,49 @@
+from unittest import mock
+
+import pytest
+
+from lightning.app.core import constants
 from lightning.app.utilities.network import find_free_network_port, LightningClient
 
 
-def test_port():
+def test_find_free_network_port():
+    """Tests that `find_free_network_port` gives expected outputs and raises if a free port couldn't be found."""
     assert find_free_network_port()
 
+    with mock.patch("lightning.app.utilities.network.socket") as mock_socket:
+        mock_socket.socket().getsockname.return_value = [0, 8888]
+        assert find_free_network_port() == 8888
+
+        with pytest.raises(RuntimeError, match="Couldn't find a free port."):
+            find_free_network_port()
+
+        mock_socket.socket().getsockname.return_value = [0, 9999]
+        assert find_free_network_port() == 9999
+
+
+@mock.patch("lightning.app.utilities.network.socket")
+def test_find_free_network_port_cloudspace(_):
+    """Tests that `find_free_network_port` gives expected outputs and raises if a free port couldn't be found when
+    cloudspace env variables are set."""
+    # Set constants
+    constants.LIGHTNING_CLOUDSPACE_HOST = "any"
+    constants.LIGHTNING_CLOUDSPACE_EXPOSED_PORT_COUNT = 10
+
+    try:
+        ports = set()
+        num_ports = 0
+
+        with pytest.raises(RuntimeError, match="All 10 ports are already in use."):
+            for _ in range(11):
+                ports.add(find_free_network_port())
+                num_ports = num_ports + 1
+
+        # Check that all ports are unique
+        assert len(ports) == num_ports
```

Will it ever raise while it is under `try` suppressing all errors/exceptions